core/src/main/java/io/kestra/scheduler/internals/StaticVNodesAssigner.java

```diff
@@ -0,0 +1,54 @@
+package io.kestra.scheduler.internals;
+
+import io.kestra.core.server.VNodeConsistentHashRing;
+import io.kestra.scheduler.SchedulerConfiguration;
+import io.kestra.scheduler.VNodesAssigner;
+import io.micronaut.context.annotation.Requires;
+import jakarta.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Simple {@link VNodesAssigner} which immediately rebalance VNodes on subscribe.
+ * <p>
+ * Only for testing-purpose.
+ */
+@Singleton
+@Requires(missingBeans = VNodesAssigner.class)
+public class StaticVNodesAssigner implements VNodesAssigner {
```

Comment:
if it's for test, move it to test sources if possible

Reply:
Micronaut seems to be lost when this class is in a test package. Could be investigated later